### PR TITLE
Add tests for EncoderFallback and DecoderFallback for UTF8Encoding

### PR DESCRIPTION
**Pull request by hughbe**
- Also adds a helper method to NegativeEncodingTests to verify
DecodeFallbackExceptions and EncoderFallback exceptions are thrown

/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 1]
```diff
@@ -256,5 +256,65 @@ public static void GetString_Invalid(Encoding encoding)
...
         }
+
+        public static unsafe void Encode_Invalid(Encoding encoding, string chars, int index, int count)
+        {
+            char[] charsArray = chars.ToCharArray();
+            byte[] bytes = new byte[encoding.GetMaxByteCount(count)];
+
+            if (index == 0 && count == chars.Length)
+            {
+                Assert.Throws<EncoderFallbackException>(() => encoding.GetByteCount(chars));
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: This test assumes the input encoding will have encoding fallback that throws exception. I know you are instantiating the UTF8 encoding with the flag does that but this also means any other encodings passed here (in the future) should do the same. I would suggest you assert encoding.EncoderFallback is equal to EncoderFallback.ExceptionFallback

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs [hunk 2]
```diff
@@ -256,5 +256,65 @@ public static void GetString_Invalid(Encoding encoding)
...
+                Assert.Throws<EncoderFallbackException>(() => encoding.GetBytes(pCharsLocal + index, count, pBytesLocal, bytes.Length));
+            }
+        }
+        public static unsafe void Decode_Invalid(Encoding encoding, byte[] bytes, int index, int count)
+        {
+            char[] chars = new char[encoding.GetMaxCharCount(count)];
+
+            if (index == 0 && count == bytes.Length)
+            {
+                Assert.Throws<DecoderFallbackException>(() => encoding.GetCharCount(bytes));
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: check the  decoder fallback is equal to DecoderFallback.ExceptionFallback

Comment (hughbe) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: done

File: src/System.Text.Encoding/tests/NegativeEncodingTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Text.Encoding/tests/NegativeEncodingTests.cs]: good idea, done

**Comment by hughbe** [comment]
Test  Innerloop Windows_NT Debug Build and Test

**Comment by tarekgh** [comment]
LGTM. Thanks.

**Pull request merged**